### PR TITLE
Added login interceptor

### PR DESCRIPTION
**Pull request by killje**
Added a login and timeout interceptor for http requests.

With this we can catch 401 responses and redirect to the login page

### CODE REVIEW

**Comment by killje** [comment]
There eslint errors, i will fix them

File: Switchboard/public/app/shared/Registry.js [hunk 1]
```diff
@@ -51,17 +53,36 @@ define(["config"], function (config) {
...
         }
     };
+	
+    /**
+     * @ngdoc method
+     * @name Registry#addFactory
+     * @description
+     * Add a factory to the angular instance
+     * 
+     * @param {String} interceptorName the name of the interceptor
```
Comment (bbusagroningen) [inline Switchboard/public/app/shared/Registry.js]: Arent we starting with upper case everywhere? Really nail picking tho!

File: Switchboard/public/app/shared/timeoutInterceptor/TimeoutInterceptor.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+define(["shared/Registry", "shared/alert/AlertFactory"], function (Registry) {
+
+    /**
+     * @ngdoc factory
+     * @module app
+     * @name timeoutInterceptor/TimeoutInterceptor
+     * @description
+     * Catches when a http response is timed out and displayes an alert message
+     * 
+     * @param {ng/service/$q} $q A angular promise object for returning a reject
```
Comment (bbusagroningen) [inline Switchboard/public/app/shared/timeoutInterceptor/TimeoutInterceptor.js]: An*

File: Switchboard/public/app/shared/timeoutInterceptor/TimeoutInterceptor.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+     * This is so other injectors can catch this even when needed
+     * @param {alert/AlertFactory} AlertFactory A alert factory, used to show a
```
Comment (bbusagroningen) [inline Switchboard/public/app/shared/timeoutInterceptor/TimeoutInterceptor.js]: An*

File: Switchboard/public/app/shared/loginInterceptor/LoginInterceptor.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	 * Catches when a http response returns an 403 unauthorized and redirects to
+	 * the login page
+	 * 
+	 * @param {ng/service/$q} $q A angular promise object for returning a reject
+	 * This is so other injectors can catch this even when needed
+	 * @param {PagesFactory} PagesFactory Uses the pages factory to get the url
+	 * for the login page
+	 * @param {page/PageFactory} PageFactory Uses the page factory to redirect
+	 * to a page
+	 * @param {alert/AlertFactory} AlertFactory A alert factory, used to show a
```
Comment (bbusagroningen) [inline Switchboard/public/app/shared/loginInterceptor/LoginInterceptor.js]: An*

File: Switchboard/public/app/shared/loginInterceptor/LoginInterceptor.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    
+	/**
+	 * @ngdoc factory
+	 * @module app
+	 * @name loginInterceptor/LoginInterceptor
+	 * @description
+	 * Catches when a http response returns an 403 unauthorized and redirects to
+	 * the login page
+	 * 
+	 * @param {ng/service/$q} $q A angular promise object for returning a reject
```
Comment (bbusagroningen) [inline Switchboard/public/app/shared/loginInterceptor/LoginInterceptor.js]: An*

**Comment by killje** [comment]
I will check in an hour, currently making dinner

**Comment by killje** [comment]
Made the changes

**Pull request merged**